drain.go

```diff
@@ -89,7 +94,8 @@ func (s *server) serveDrain(w http.ResponseWriter, r *http.Request) {
 	batchCounter.Inc(1)
 
 	parseStart := time.Now()
-	lp := lpx.NewReader(bufio.NewReader(r.Body))
+	buf := bytes.Buffer{}
+	lp := lpx.NewReader(bufio.NewReader(io.TeeReader(r.Body, &buf)))
```

Comment:
We need to keep all the headers and such, which this doesn't currently do. But, this was more or less the approach I'd like to see.

Guard shadowUrls, provide an HTTP interface to updating them and removing them, which we then can use deploymaster to add / remove stuff. Then, do the env var to catch restarts.